Update initializer to work with ember 2.1.x

gregjopa opened the pull request:
This change adds support for the new 2.1.x container api. I've tested this change in 2.1.0, 2.0.0, and 1.13.7.

I'm not sure this is the best way to handle this api change. Let me know if you have any suggestions on how to better do this and I will update the PR.

This initializer code was originally added by @Dhaulagiri.

gregjopa commented:
I looked into these test failures and I'm not sure how to fix this. From what I've researched, the container api is not stable. The ember source code has the following message in it: "The public API for `Container` is still in flux and should not be considered stable". https://github.com/emberjs/ember.js/blob/master/packages/container/lib/container.js#L12-L13

Its not clear to me what this register container code does. Is there any way to update this initializer to use `application.inject()` instead of `container.registerOptionsForType()`? I'd like to help but I'm lost on this one.

ahmadsoe commented:
This initializer used to register the default chart options `app/highcharts-configs/application.js` to container.  But I think ember CLI has registered them by default using ember-resolver,  so there is no need for the initializer.

gregjopa commented:
I tried deleting the initializer and updating the tests but the tests fail and the dummy app doesn't render the charts. I tested this with ember-cli 1.13.8. Do you know of any other addons that use initializers? 

I haven't seen any other addons inject settings into the container. Since the container api is not stable perhaps we should rethink how default chart options are loaded.

ahmadsoe commented:
I can't think another way how to load default chart option. `ember-simple-auth`,  `ember-cli-mirage` or `ember-pop-over` inject their configs into the container. 

Have you tried to change the lookup method here https://github.com/ahmadsoe/ember-highcharts/blob/master/addon/utils/option-loader.js#L6
because the default chart config is just a simple function, not an Ember class, it requires `container.lookupFactory` instead of `container.lookup`.

gregjopa commented:
Thanks! I deleted the initializer and updated the chart config loader to use `container.lookupFactory` and everything seems to be working. I pushed my changes and all the tests passed.

ahmadsoe commented on tests/dummy/app/app.js:
```diff
@@ -13,6 +12,4 @@ App = Ember.Application.extend({
   Resolver: Resolver
 });
 
-loadInitializers(App, config.modulePrefix);
```
is this necessary?

gregjopa replied:
Good catch. I reverted that change. Now the app.js file matches exactly what ember-cli defaults for app.js with a new project:

```javascript
import Ember from 'ember';
import Resolver from 'ember/resolver';
import loadInitializers from 'ember/load-initializers';
import config from './config/environment';

var App;

Ember.MODEL_FACTORY_INJECTIONS = true;

App = Ember.Application.extend({
  modulePrefix: config.modulePrefix,
  podModulePrefix: config.podModulePrefix,
  Resolver: Resolver
});

loadInitializers(App, config.modulePrefix);

export default App;
```

ahmadsoe commented:
This looks great @gregjopa. Thank you.

gregjopa commented:
Thanks for merging!

flaviooliveri commented:
Thanks guys.
